src/storage/src/monitor/local_metrics.rs

```diff
@@ -74,78 +96,41 @@ impl StoreLocalStatistic {
         self.cache_meta_block_miss += local_cache_meta_block_miss;
     }
 
-    pub fn report(&self, metrics: &HummockStateStoreMetrics, table_id_label: &str) {
-        if self.cache_data_block_total > 0 {
-            metrics
-                .sst_store_block_request_counts
-                .with_label_values(&[table_id_label, "data_total"])
-                .inc_by(self.cache_data_block_total);
-        }
-
-        if self.cache_data_block_miss > 0 {
-            metrics
-                .sst_store_block_request_counts
-                .with_label_values(&[table_id_label, "data_miss"])
-                .inc_by(self.cache_data_block_miss);
-        }
-
-        if self.cache_meta_block_total > 0 {
-            metrics
-                .sst_store_block_request_counts
-                .with_label_values(&[table_id_label, "meta_total"])
-                .inc_by(self.cache_meta_block_total);
-        }
-
-        if self.cache_meta_block_miss > 0 {
-            metrics
-                .sst_store_block_request_counts
-                .with_label_values(&[table_id_label, "meta_miss"])
-                .inc_by(self.cache_meta_block_miss);
-        }
-
+    fn report(&self, metrics: &mut LocalStoreMetrics) {
```

Comment:
Can we pass the self ownership instead of ref, to avoid duplicate reporting ?

Reply:
It seems difficult because we can not pass the ownership of local_stats because it is a member of `HummockStorageIterator`

Reply:
I would give a guard to make sure that this function would be call exactly once.